src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -258,8 +258,6 @@
                                              (set-input-state
                                               #(controlled-input/->crypto % conversion-rate)))
                                            (set-crypto-currency (not crypto-currency?)))]
-    (rn/use-unmount
-     #(rf/dispatch [:wallet/clean-up-transaction-flow]))
```

Comment:
On send flow (`To Address -> Asset -> Amount`), if you navigate back from the input amount, it causes a minor recursion bug where the user goes back to Asset and upon selecting the asset, the user is taken to selecting address again as the to address is cleaned up.

Reply:
mm good point :+1: . And there's no other screen we can put this in, since the flows can have different entry points. Hopefully we can figure out a way to structure these flows/screens to be able to clean-up on unmount eventually, so we no longer rely on delayed clean-ups when done.

Reply:
I agree, @clauxx. It's a bit tricky. It's not just the delayed cleanup but also the different clean-up events scattered across the screens/flows. We will research this delayed approach and create a solution post 2.32 release 👍